### PR TITLE
Update schema for DQ framework

### PR DESCRIPTION
**Pull request by EuiYeonJang**
This PR updates the following schemas to include the necessary data quality framework scores and updates the `generate_table` function with the logic necessary to populate those columns:
- `companies_datamodel` to include `data_granularity` score
- `tiltLedger_mapping_datamodel` (newly created for the datamodel layer)
- `tiltLedger_datamodel` to include `model_certainty` score
- `sources_mapper_datamodel` to include `data_source_reliability` score

### CODE REVIEW

File: functions/processing_datamodel.py [hunk 1]
```diff
@@ -146,86 +153,149 @@ def generate_table(table_name: str) -> None:
...
+            .withColumn(
+                "product_name",
+                F.when(F.col("product_name") == "", None).otherwise(
+                    F.col("product_name")
+                ),
+            )
+            .orderBy("company_id")
+        )
+
+        companies_raw_final.data = companies_raw_final.data.join(
```
Comment (EuiYeonJang) [inline functions/processing_datamodel.py:258]: Could you please pay a special attention this line? If I use the custom_join, the code errors due to the mapping column missing

**Comment by EuiYeonJang** [comment]
Please be advised: unfortunately, a lot of changes reflected in the files include formatting changes

File: functions/processing_datamodel.py (hunk 1 above)
Comment (SKruthoff) [reply functions/processing_datamodel.py:258]: Yes the map column is going missing due to the group by function. I have also implemented a custom_average function for the code that Sohail needs. The issue here is that in an aggregation you would not only want to concat the strings, but you would also want to concatenate the values in the map column that you are grouping by. Do you want to take a shot at implementing it yourself with some examples/ideas from my side?

**Comment by SKruthoff** [comment]
Can you maybe point me to the code that needs to be reviewed specifically? I see no flaw in the code, but the changes in formatting make it hard to check it properly.

File: functions/custom_dataframes.py [hunk 2]
```diff
@@ -513,9 +612,83 @@ def custom_drop(self, columns: list):
...
-        return CustomDF(self._name, self._spark_session, copy_df, self._partition_name, self._history)
+        return CustomDF(
+            self._name,
+            self._spark_session,
+            copy_df,
+            self._partition_name,
+            self._history,
+        )
+
+    def custom_concatenate(
```
Comment (SKruthoff) [inline functions/custom_dataframes.py:623]: Is this still relevant?

**Review by SKruthoff — CHANGES_REQUESTED** [review]
In principle I agree that this PR is correct. I would again check that all parts that are contained in this PR are still relevant. Specifically the custom_concatenate can be removed I think.

**Review by SKruthoff — APPROVED** [review]
Looks good to me!

**Pull request merged**